agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java

```diff
@@ -44,11 +44,13 @@ public TestRunner(AgentManagementService agentManagementService, TestTaskRunCall
         this.testRunDeviceOrchestrator = testRunDeviceOrchestrator;
     }
 
-    public void runTestOnDevice(TestTask testTask, TestRunDevice testRunDevice, Logger logger) {
+    public void runTestOnDevice(TestTask testTask, TestRunDevice testRunDevice) {
         checkTestTaskCancel(testTask);
-        logger.info("Start running tests {}, timeout {}s", testTask.getTestSuite(), testTask.getTimeOutSecond());
+        if (testRunDevice.getLogger() != null) {
```

Comment:
Do we need this condition? If return false, do we need to set logger again?

Reply:
Make sense, I can turn this into an assertion, claiming logger is required.

Reply:
Updated, resolve.